zomes/wiki/code/src/section.rs

```diff
@@ -12,51 +12,43 @@ use hdk::holochain_core_types::{
 };
 use hdk::holochain_json_api::{error::JsonError, json::JsonString};
 use hdk::holochain_persistence_api::cas::content::Address;
-use hdk::prelude::Entry::App;
+use hdk::prelude::*;
 use hdk::{
     entry_definition::ValidatingEntryType,
     error::ZomeApiResult,
     //,
     // AGENT_ADDRESS, AGENT_ID_STR,
 };
 
-use crate::page;
 use crate::page::Page;
 use crate::utils::validate_agent_can_edit;
 
 #[derive(Serialize, Deserialize, Debug, DefaultJson, Clone)]
 pub struct Section {
-    page_address: Address,
-    r#type: String,
-    content: String,
-    rendered_content: String,
-    timestamp: String,
-}
-#[derive(Serialize, Deserialize, Debug, DefaultJson, Clone)]
-pub struct Section2 {
+    anchor_address: Option<Address>,
     r#type: String,
     content: String,
     rendered_content: String,
     timestamp: String,
 }
 
 impl Section {
-    pub fn from(section: Section2, page_address: Address) -> Section {
+    pub fn from(self, anchor_address: Address) -> Section {
         Section {
-            page_address,
-            r#type: section.r#type,
-            content: section.content,
-            rendered_content: section.rendered_content,
-            timestamp: section.timestamp,
+            anchor_address: Some(anchor_address),
+            r#type: self.r#type,
+            content: self.content,
+            rendered_content: self.rendered_content,
+            timestamp: self.timestamp,
         }
     }
     pub fn entry(self) -> Entry {
-        App("pageElement".into(), self.into())
+        Entry::App("pageSection".into(), self.into())
```

Comment:
Change `pageSection` to `section` or `page_section` to be consistent with rust snake_case